fluss-client/src/main/java/com/alibaba/fluss/client/admin/FlussAdmin.java

```diff
@@ -277,6 +279,21 @@ public CompletableFuture<List<PartitionInfo>> listPartitionInfos(TablePath table
                 .thenApply(ClientRpcMessageUtils::toPartitionInfos);
     }
 
+    @Override
+    public CompletableFuture<List<PartitionInfo>> listPartitionInfos(
+            TablePath tablePath, PartitionSpec partitionSpec) {
```

Comment:
We introduce a new interface instead of modifying the exists interface is to ensure backward compatibility, but internally class we can merge the implementation with `CompletableFuture<List<PartitionInfo>> listPartitionInfos(TablePath tablePath)`. Additionally, in my opinion, the interface `CompletableFuture<List<PartitionInfo>> listPartitionInfos(TablePath tablePath)` should be marked as deprecated.